### PR TITLE
[MRG] Better molecule type checks

### PR DESCRIPTION
**Pull request by luizirber**
This triggers bug #781, still need to fix it

## Checklist

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
This is missing `tests/test-data/protein_781.sig` :)

**Comment by ctb** [comment]
(added the signature from @bluegenes example)

**Comment by luizirber** [comment]
so, I looked into this and... Currently we have two types of preprocessing of sequences (before adding to minhash): DNA and protein. For proteins, we can do three encodings: aminoacid, dayhoff and hp. So, something like this:
- DNA
- protein
  - aminoacid
  - dayhoff
  - hp

The main issue for this PR is that `load_signatures` with a `select_moltype=protein` will also bring the `hp` encoding, on top of the aminoacid encoding (the 'default' for protein). I think we should break that hierarchy and do one `moltype` per encoding (even if aa, dayhoff and hp are all `protein`).

This is also very relevant for #751.

**Comment by luizirber** [comment]
The good thing is that `molecule` is always one of `dna`, `protein`, `dayhoff` or `hp` in the sig JSON file, so no changes needed there.

We would need to change `kmer_min_hash.hh`, because nowadays `is_protein` is true for `dayhoff` and `hp` too (and it should be only true for `aa`?)

**Comment by luizirber** [comment]
(also pinging @olgabot and @pranathivemuri to see if this breaks any of their pipelines)

File: tests/test_bugs.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from __future__ import print_function, unicode_literals
+from . import sourmash_tst_utils as utils
+
+def test_bug_789():
```
Comment (pranathivemuri) [inline tests/test_bugs.py]: bug 789 or 781? what is this number actually?

Comment (pranathivemuri) [reply tests/test_bugs.py]: Just saw that it's an issue above.

**Comment by pranathivemuri** [comment]
The changes look good to me. I don't see how it might break a pipeline as long as all the tests are passing.

**Comment by luizirber** [comment]
> The changes look good to me. I don't see how it might break a pipeline as long as all the tests are passing.

So.. there is no way of making the tests pass with the changes I proposed. This is the main culprit:
https://github.com/dib-lab/sourmash/blob/57f6efc7dfe298e521529549e588c4745d0bb159/tests/test_sourmash_compute.py#L496

The main change in this PR is making `is_protein` represent only the `protein` molecule type (the original aa translation), but for this test the `x.minhash.is_molecule_type('protein')` returns True for both `protein` and `hp` molecule types. That's why I'm asking if this breaks any pipelines, because if it doesn't we change it from `== 4` to `== 2`, and release `2.3.1`.

**Comment by luizirber** [comment]
> The main change in this PR is making `is_protein` represent only the `protein` molecule type (the original aa translation), but for this test the `x.minhash.is_molecule_type('protein')` returns True for both `protein` and `hp` molecule types. That's why I'm asking if this breaks any pipelines, because if it doesn't we change it from `== 4` to `== 2`, and release `2.3.1`.

I went ahead changed the test value. That should mean a major version bump, but since mostly @olgabot and team are using this feature I think we should treat as a behavior bug instead and release a quick fix.

Ready for review and merge @ctb @olgabot

**Pull request merged**